tests/unit/document_tests.py

```diff
@@ -523,9 +524,18 @@ def test_update_field_success_on_retry(self):
         # Mock when saving the document
         # 1st call throw a 409
         # 2nd call delegate to the real doc.save()
-        with mock.patch('cloudant.document.Document.save',
-                        side_effect=[requests.HTTPError(response=mock.Mock(status_code=409, reason='conflict')),
-                                     doc.save()]) as m_save:
```

Comment:
Oh I see, the .save() was happening before the mock was even initialised. I guess this bad test properly failed in your pr #429 since the save didn't execute given there are no local document changes at this point.

Reply:
I guess `doc.save` (not`doc.save()`) would be less wrong. But that would just call the mock again.

Reply:
I've tried this way but the problem is side_effect as a list raise exceptions or return values. So in this case it would effectively return the callable doc.save instead of running it. And of course, because how the patch is made yes, it would loop.

Reply:
Yeah, just read the docs. I assumed wrong.